chain-test/src/test/scala/org/bitcoins/chain/blockchain/ChainHandlerTest.scala

```diff
@@ -19,4 +27,34 @@ class ChainHandlerTest extends ChainUnitTest {
       foundHeaderF.map(found => assert(found.get == newValidHeader))
   }
 
+  it must "peer with bitcoind via zmq and have blockchain info relayed" in withBitcoindZmqChainHandler {
+    case bitcoindChainHandler: BitcoindChainHandler =>
+      val bitcoind = bitcoindChainHandler.bitcoindRpc
+
+      val chainHandler = bitcoindChainHandler.chainHandler
+
+      val assert1F = chainHandler.getBlockCount
+        .map(count => assert(count == 0))
+
+      //mine a block on bitcoind
+      val generatedF = assert1F.flatMap(_ => bitcoind.generate(1))
+
+      generatedF.flatMap { headers =>
+        val hash = headers.head
+        val foundHeaderF: Future[Unit] = {
+          //test case is totally async since we
+          //can't monitor processing flow for zmq
+          //so we just need to await until we
+          //have fully processed the header
+          RpcUtil.awaitConditionF(() =>
+            chainHandler.getHeader(hash).map(_.isDefined))
```

Comment:
nit: Intermediate val here and a more descriptive name than `assert1F` such as `noBlocksF` would be nice :)